Assets/Scripts/NPCs/Flocking/Behaviour/AvoidanceBehaviour.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace NPCs.Flocking.Behaviour
+{
+    [CreateAssetMenu(menuName = "Flock/Behaviour/Avoidance")]
+
+//Move agent away from the neighbour agents
+    public class AvoidanceBehaviour : FilteredFlockBehavior
+    {
+        public override Vector3 CalculateMove(NPCManager npc, List<Transform> context, FlockManager flock)
+        {
+            Vector3 avoidanceMove = Vector3.zero;
+            int nAvoid = 0;
+
+            var filteredContext = FilterContext(context, npc);
+            
+            if (filteredContext.Count == 0)
+            {
+                return Vector3.zero;
+            }
+
+            foreach (Transform item in filteredContext)
+            {
+                nAvoid++;
+                avoidanceMove += (npc.transform.position - item.position);
+            }
+
+            if (nAvoid > 0)
+            {
+                avoidanceMove /= nAvoid;
+            }
+
+            return avoidanceMove;
+        }
+
+        private List<Transform> FilterContext(List<Transform> context, NPCManager npc)
+        {
+            List<Transform> filteredContext = (filter == null) ? context : filter.Filter(npc, context);
+
+            filteredContext = (filter == null)
+                ? filteredContext
+                : RemoveElementFromContextByName(filteredContext, "Ground");
+
+            filteredContext = (filter == null)
+                ? filteredContext
+                : RemoveElementFromContextByName(filteredContext, "RightHand");
+
+            return filter == null ? filteredContext : RemoveItSelfFromContext(filteredContext, npc);
+        }
+    }
```

Comment:
This is because the different forces don't deal with the same objects. For instance, steering behaviour only takes into account NPCs, so we just need to filter out the right hand since Ground is already not taken into account. This is the same thing for cohesion. In avoidance, we need to take into account both the NPCs and the environment individually, as we need to handle the cases where the character has to avoid other NPCs and the case where the NPC has to avoid the environment.